channeldb/invoices.go

```diff
@@ -423,6 +429,18 @@ func deserializeInvoice(r io.Reader) (*Invoice, error) {
 		return nil, err
 	}
 
+	// Before we return with the current invoice, we'll check to see if
+	// there's still enough space in the buffer to read out the final ctlv
+	// delta. We'll get an EOF error if there isn't any thing else
+	// lingering in the buffer.
+	err = binary.Read(r, byteOrder, &invoice.Terms.FinalCltvDelta)
```

Comment:
More than sure that you are aware of this, but such approach might be not be suitable for future, because:
* New fields could be added only at the end struct
* If we have two people changing the struct simultaneously they might affect each other.

Instead I believe we could add versioning in serialisation/deserialisation functions, and add simple migration, which decodes struct with previous version and encodes with new one.

Reply:
Yeah it was meant as kinda of the "slim path" w.r.t to the set of changes. The alternative, was to simply decode the payreq already in the invoice, but then that would've required modifications to the switch, whereas I wanted to keep the set of changes isolated (which in hindsight was a mistake). We use this pattern in a few other areas when we _know_ it's always at the end of the struct, but this isn't the case here...

If two people are changing it, well then they'll need to handle the git merge conflicts as normal. 

IMO there's no need for a version, just a migration as we only want to maintain a single version of each struct within the database.

Reply:
> IMO there's no need for a version, just a migration as we only want to maintain a single version of each struct within the database.

Yeah, but when I was writing migration in this [PR](https://github.com/lightningnetwork/lnd/pull/1452) I found that in order to not duplicate deserialisation/serialisation functions and at the same time be able to test migration I need to put version in function.

Reply:
Well the migration tests can just manually replicate the existing prior version. The migrations themselves will only rarely be run, and a lil duplication never hurt nobody ;)

Reply:
😎